client/app/reader/actions.js

```
@@ -43,34 +47,148 @@ export const onScrollToComment = (scrollToComment) => ({
+      then(_.noop);
+
+    dispatch({
+      type: Constants.REQUEST_DELETE_ANNOTATION,
+      payload: {
+        annotationId
+      }
+    });
+  };
+
+export const requestMoveAnnotation = (annotation) => (dispatch) => {
+  dispatch({
+    type: Constants.REQUEST_MOVE_ANNOTATION,
+    payload: {
+      annotation
+    }
+  });
+
+  const data = ApiUtil.convertToSnakeCase({ annotation });
+
+  ApiUtil.patch(`/document/${annotation.documentId}/annotation/${annotation.id}`, { data }).then(_.noop);
+};
+
+export const cancelEditAnnotation = (annotationId) => ({
+  type: Constants.CANCEL_EDIT_ANNOTATION,
+  payload: {
+    annotationId
+  }
+});
+export const updateAnnotationContent = (annotationId, content) => ({
+  type: Constants.UPDATE_ANNOTATION_CONTENT,
   payload: {
-    state: null
+    annotationId,
+    content
   }
 });
+export const updateNewAnnotationContent = (content) => ({
+  type: Constants.UPDATE_NEW_ANNOTATION_CONTENT,
+  payload: {
+    content
+  }
+});
+
+export const requestEditAnnotation = (annotation) => (dispatch) => {
+  dispatch({
+    type: Constants.REQUEST_EDIT_ANNOTATION,
+    payload: {
+      annotationId: annotation.id
+    }
+  });
 
-export const handleSelectCommentIcon = (comment) => ({
-  type: Constants.SCROLL_TO_SIDEBAR_COMMENT,
+  const data = ApiUtil.convertToSnakeCase({ annotation });
+
+  ApiUtil.patch(`/document/${annotation.documentId}/annotation/${annotation.id}`, { data }).then(_.noop);
+};
+
+export const startPlacingAnnotation = () => ({ type: Constants.START_PLACING_ANNOTATION });
+
+export const placeAnnotation = (pageNumber, coordinates, documentId) => ({
+  type: Constants.PLACE_ANNOTATION,
   payload: {
-    scrollToSidebarComment: comment
+    page: pageNumber,
+    x: coordinates.xPosition,
+    y: coordinates.yPosition,
+    documentId
   }
 });
 
+export const stopPlacingAnnotation = () => ({ type: Constants.STOP_PLACING_ANNOTATION });
+
+export const createAnnotation = (annotation) => (dispatch) => {
```

ok I misunderstood the order (I thought you typed the comment then dragged). Not sure about this from a UX perspective (we should provide error feedback rather than silently fail), but sounds good for now